### DISCUSSION TITLE
How to replace __mapper_args__ order_by in sqlalchemy 2

### DISCUSSION THREAD

**Discussion by LeResKP** [sqlalchemy/sqlalchemy#12010] [Usage Questions] [answered]
Hello,

I'm using sqlalchemy 1.3.13 and want to migrate the sqlalchemy 2 but I'm not sure the best way to replace the order_by of \_\_mapper_args__.

When querying the relationship, that's okay, there is an order_by on relationship like in the following example.

But when querying a table directly I didn't found a way to set a default order_by on the model.

If you check the following code, I added an order by on the Address \_\_mapper_args__ to that there is a default order_by on each query.

What is the best/nicer way to keep this behaviour in sqla 2 ?

Thanks in advance for you help !
Aurélien

```
from sqlalchemy import Column, ForeignKey, Integer, String, create_engine
from sqlalchemy.ext.declarative import declarative_base, declared_attr
from sqlalchemy.orm import relationship, sessionmaker

Base = declarative_base()
engine = create_engine("sqlite://", echo=False)

Session = sessionmaker(engine)
session = Session()


class User(Base):
    __tablename__ = "user"

    id = Column(Integer, primary_key=True)
    name = Column(String)
    fullname = Column(String)
    nickname = Column(String)

    addresses = relationship("Address", backref="user", order_by="Address.id")


class Address(Base):
    __tablename__ = "address"

    id = Column(Integer, primary_key=True)
    user_id = Column(ForeignKey("user.id"))
    email_address = Column(String)

    @declared_attr
    def __mapper_args__(cls):
        return {
            "order_by": cls.id,
        }


Base.metadata.create_all(engine)

print(session.query(Address))
# SELECT address.id AS address_id, address.user_id AS address_user_id, address.email_address AS address_email_address 
FROM address ORDER BY address.id
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

That property was deprecated in 1.1 and it was removed in 1.4. You will need to manually add an order by to your queries

**Reply by LeResKP** [reply]
I was afraid about this answer since I didn't saw anything comparable in the doc. Thanks for you answer!
